integrationtests/pom.xml

```diff
@@ -125,6 +137,14 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>18</source>
```

Comment:
I imagine this is to get it working in your IDE? I also often go into this pom and do something similar to get it running in IDEA :cry: , need to try setting it to the experimental project mode.

You could instead set:
```
<properties>
    <java.version>19</java.version>
</properties>
```

Since the test version is 19 already and I don't think there's harm in raising the integrationtest language level to 19 for `main`.